src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (
+      prey instanceof Herbivore
+      && !prey.hidden
+    ) {
+      prey.health -= 50;
+
+      if (prey.health === 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal !== prey);
+      }
```

Comment:
Let's avoid nested if's

Reply:
And your solution will be more scalable if you'll check if health is greater or equal than 0
What if we'll change damage or health?